tracing/src/main/java/com/palantir/tracing/CloseableTracer.java

```diff
@@ -81,6 +81,7 @@ public static <T> CloseableTracer startSpan(
     }
 
     @Override
+    @SuppressWarnings("DesignForExtension")
     public void close() {
```

Comment:
Given the constructor is package protected, this is probably fine. May be worth comment that subtypes either need to call super to complete span, or if they completely override `close()` they are responsible for completing the span.

Reply:
FWIW I'm basically treating `CloseableTracer` like an interface. Maybe I should just make it an abstract class - I don't think that's an ABI break. There are no external subclasses because the constructor was private (and anyone who is subclassing this is doing something they shouldn't be).